soh/src/overlays/actors/ovl_En_Ds/z_en_ds.c

```diff
@@ -195,17 +196,19 @@ void EnDs_OfferBluePotion(EnDs* this, PlayState* play) {
                         this->actionFunc = EnDs_TalkNoEmptyBottle;
                         return;
                     case 2: // have 100 rupees and empty bottle
-                        Rupees_ChangeBy(-100);
+                        if(!GameInteractor_Should(VB_RANDO_GRANNY_TAKE_MONEY, false, this)){
+                            Rupees_ChangeBy(-100);
+                        }
```

Comment:
This is confusing with the double inversion, as well, the hooks shouldn't have specific references to rando as they can be used for other purposes in the future.
```suggestion
                        if (GameInteractor_Should(VB_GRANNY_TAKE_MONEY, true, this)) {
                            Rupees_ChangeBy(-100);
                        }
```

Reply:
I was double negativing because I thought I was supposed to return true if I did something, specifically, if rando granny took money, in which case the vanilla take money should be skipped. If this is wrong I can change it.

Reply:
The main idea is that the hook returns true if the hook name should happen, in this case we're checking if granny should take the money, in the case where this is handled by rando we'd return false to make the original code not run. We're checking: `Should GRANNY_TAKE_MONEY?`